### PR TITLE
vcsim: add guest operations process support

### PR DESCRIPTION
**Pull request by dougm**
The original plan was to run govmomi/toolbox inside the container.
  Pros: toolbox uses Go's stdlib for process and file ops, no shelling out and parsing command output.
  Cons: would require binary added to containers, manage the process lifecycle and transport (e.g. unix socket).
With this change, we instead use a hybrid of the toolbox.ProcessManager and docker exec.
  Pros: negates the cons listed above.
  Cons: shelling out to docker exec and having to parse output in some cases (e.g. ListFiles)

### CODE REVIEW

**Comment by dougm** [comment]
#2375 using here will avoid the compile errors

**Comment by embano1** [comment]
Let me know when this is good for review

File: simulator/guest_operations_manager.go [hunk 1]
```diff
@@ -108,4 +115,327 @@ func (m *GuestFileManager) InitiateFileTransferFromGuest(ctx *Context, req *type
...
+
+		_, fault := vm.run.exec(ctx, vm, req.Auth, args)
+		if fault != nil {
+			body.Fault_ = Fault("", fault)
+			return body
+		}
+	}
+
+	if attr.GroupId != nil || attr.OwnerId != nil {
+		// TODO: chown
```
Comment (scottballenger) [inline simulator/guest_operations_manager.go:435]: Should this generate a fault (not implemented) or log a warning?

**Review by scottballenger — APPROVED** [review]
LGTM Doug. Thanks for the changes!

File: simulator/container.go [hunk 2]
```diff
@@ -264,17 +280,34 @@ func (c *container) stop(vm *VirtualMachine) {
...
+}
+
+// restart the container (if any) for the given vm.
+func (c *container) restart(ctx *Context, vm *VirtualMachine) {
+	if c.id == "" {
+		return
+	}
+
+	cmd := exec.Command("docker", "restart", c.id)
+	err := cmd.Run()
```
[the remark this answers is not in the document]
Comment (dougm) [reply simulator/container.go:310]: Good call, done.

File: simulator/guest_operations_manager.go (hunk 1 above)
Comment (dougm) [reply simulator/guest_operations_manager.go]: Added mapping to chmod + chgrp commands.

**Review by scottballenger — APPROVED** [review]

**Review by scottballenger — APPROVED** [review]

**Pull request merged**